core/Db/TransactionLevel.php

```diff
@@ -54,7 +59,13 @@ public function setUncommitted()
             $this->db->query('SET SESSION TRANSACTION ISOLATION LEVEL READ UNCOMMITTED');
             $this->statusBackup = $backup;
```

Comment:
Shouldn't the `statusBackup` only be set if it wasn't set before? Just wondering what would happen when calling the method twice, without calling `restorePreviousStatus` before. In that case the second call would overwrite the `statusBackup` with `READ UNCOMMITTED` and the previously set level gets lost.

Reply:
@sgiehl i was wonder the same thing. Otherwise the restore will be always skipped. I am right.

Reply:
This would be a different issue overall maybe and not as part of this. Generally, because we create each time a separate instance of the transaction level it should work and be fine. Like 

```php
$t = new TransactionLevel();
$t->setUncommitted();
....
$x = new TransactionLevel();
$x->setUncommitted();
$x->restorePrevious();
...
$t->restorePrevious();
```

AFAIK `$t` should then restore the previous one again correctly. `$x` would technically not need to try and set the status if `backup` is already `READ UNCOMMITTED`.